qiskit/circuit/quantumcircuit.py

```diff
@@ -904,6 +904,8 @@ def compose(
             )
 
         dest = self if inplace else self.copy()
+        dest.duration = None
+        dest.unit = "dt"
```

Comment:
I added it here because this used to happen as a result of calls to `QuantumCircuit._append`, and we now use `CircuitScopeInterface.extend`.

In theory, perhaps updating these _should_ belong in `_OuterCircuitScopeInterface.extend`, since the original behavior would be to set these only if the scope is the root `QuantumCircuit`. Thoughts?

Are there any cases you see where these should have been set before in `compose` but weren't?

Reply:
tbh, I don't really know - I don't know how these attributes are meant to work properly.